test/src/test/java/jenkins/install/UpgradeWizardTest.java

```diff
@@ -0,0 +1,86 @@
+package jenkins.install;
+
+import hudson.util.VersionNumber;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Kohsuke Kawaguchi
+ */
+public class UpgradeWizardTest {
+    @Rule
+    public final JenkinsRule j = new JenkinsRule();
+
+    @Inject
+    UpgradeWizard uw;
+
+    @Before
+    public void setup() {
+        j.jenkins.getInjector().injectMembers(this);
+    }
+
+    @Test
+    public void snooze() throws Exception {
+        assertTrue(uw.isDue());
+        uw.doSnooze();
+        assertFalse(uw.isDue());
+    }
+
+    /**
+     * If not upgraded, the upgrade should cause some side effect.
+     */
+    @Test
+    public void upgrade() throws Exception {
+        assertTrue(j.jenkins.getUpdateCenter().getJobs().size() == 0);
+        assertTrue(newInstance().isDue());
+        assertNotSame(UpgradeWizard.NOOP, uw.doUpgrade());
+
+        // can't really test this because UC metadata is empty
```

Comment:
I don't know if I want a test to download & install a plugin, so I'm actually happy as is.